source/module_pw/pw_basis_k.cpp

```diff
@@ -90,7 +91,12 @@ void PW_Basis_K::setupIndGk()
         {
             this->npwk_max = ng;
         }
+        if ( npwk_min > ng)
+        {
+            npwk_min = ng;
+        }
     }
+    ModuleBase::CHECK_WARNING_QUIT((npwk_min==0), "PW_Basis_K::setupIndGk", "some cores have no plane waves!");
```

Comment:
We only use pool_world for pw_basis class.
pw_basis.h:
![image](https://user-images.githubusercontent.com/76200646/186110697-7a40565a-36da-420b-8f76-3b41ea3961c4.png)

Reply:
Did pool_world different with MPI_COMM_WORLD?
We should terminate program in all processors but not only one k pool when kpar>1.